cpp/src/arrow/csv/reader.cc

```diff
@@ -742,6 +751,25 @@ class ReaderMixin {
     return ParseResult{std::move(parser), static_cast<int64_t>(parsed_size)};
   }
 
+  /**
+   * @brief add the `size()` of all buffers in `block` to `parser_offset_`
+   * @return the parser offset which should be used by the caller
+   */
+  int64_t UpdateParserOffset(const CSVBlock& block) {
+    parser_offset_ += block.bytes_skipped;
+    auto result = parser_offset_;
```

Comment:
It's been a hard habit for me to break but `result` is not a good variable name.  Maybe `updated_offset` or something like that?